src/content/docs/en/guides/upgrade-to/v3.mdx

```diff
@@ -0,0 +1,531 @@
+
+Astro v3.0 changes the [default port](/en/reference/cli-reference/#--port-number) to `4321`. 🚀
+
+#### What should I do?
+
+Update any existing references to `localhost:3000`, for example in tests or in your `README`, to reflect the new port `localhost:4321`.
+
+
+### Changed default: import.meta.env.BASE_URL `trailingSlash`
+
+In Astro v2.x, `import.meta.env.BASE_URL` appended your [`base`](/en/reference/configuration-reference/#base) setting with a [trailingSlash](/en/reference/configuration-reference/#trailingslash) by default. `trailingSlash: "ignore"` also appended a trailing slash.
+
+Astro v3.0 no longer appends `import.meta.env.BASE_URL` with a trailing slash by default, nor when `trailingSlash: "ignore"` is set. (The existing behavior of `base` in combination with `trailingSlash: "always"` or `trailingSlash: "never"` is unchanged.)
+
+#### What should I do?
+
+If your `base` already has a trailing slash, no change is needed.
+
+If your `base` does not have a trailing slash, add one if you wish to preserve the previous default (or `trailingSlash: "ignore"`) behavior:
+
+```js title="astro.config.mjs" del={4} ins={5}
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
+  base: 'my-base',
+  base: 'my-base/',
+});
+```
+
+### Changed default: `compressHTML`
+
+In Astro v2.x, Astro only compressed your emitted HTML when [`compressHTML`](/en/reference/configuration-reference/#compresshtml) was explicitly set to `true`. The default value was `false`.
+
+Astro v3.0 now compresses emitted HTML by default.
+
+#### What should I do?
+
+You can now remove `compressHTML: true` from your configuration as this is the new default behavior.
+
+```js title="astro.config.mjs" del={4}
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
+	compressHTML: true
+})
+```
+
+You must now set `compressHTML: false` to opt out of HTML compression.
+
+### Changed default: `scopedStyleStrategy`
+
+In Astro v2.x, the default value of [`scopedStyleStrategy`](/en/reference/configuration-reference/#scopedstylestrategy) was `"where"`.
+
+Astro v3.0 introduces a new, default value: `"attribute"`. By default, styles are now applied using `data-*` attributes.
+
+#### What should I do?
+
+To retain your project's current [style scoping](/en/guides/styling/#scoped-styles), update the configuration file to the previous default value: 
+
+
+```js title="astro.config.mjs" ins={4}
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
+	scopedStyleStrategy: "where"
+})
+```
+
+### Changed default: `inlineStyleSheets`
+
+In Astro v2.x, all project stylesheets were sent as link tags by default. You could opt in to inlining them into `<style>` tags everytime with `"always"`, or to inlining only stylesheets below a certain size with `"auto"` by setting the [`build.inlineStylesheets`](/en/reference/configuration-reference/#buildinlinestylesheets) configuration. The default setting was `"never"`.
```

Comment:
Please write "every time" as two words. As one word, it would become an adjective and we don't use it nearly as often as we use the adjective "everyday".